client/packages/system/src/Patient/JsonForms/components/DispendedDuration.tsx

```diff
@@ -119,6 +142,11 @@ const UIComponent = (props: ControlProps) => {
   );
   const error = !!errors;
 
+  useEffect(() => {
+    if (options) {
+      setLocalData(extractProperty(data, options.targetField) ?? 0);
+    }
+  }, [options]);
```

Comment:
```suggestion
  }, [data, options]);
```

doesn't the effect also depend on data? ( and `setLocalData` and `extractProperty` but I am not too concerned about including the functions )

in which case it could be included in the following effect?

which could also be checking for a valid `options` object now?

Reply:
Ok done thanks!